x-pack/dockerlogbeat/pipelineManager/clientLogReader.go

```diff
@@ -0,0 +1,125 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package pipelinemanager
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/docker/engine/api/types/plugins/logdriver"
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/publisher/pipeline"
+	pb "github.com/gogo/protobuf/io"
+	"github.com/pkg/errors"
+	"github.com/tonistiigi/fifo"
+)
+
+// ClientLogger is an instance of a pipeline logger client meant for reading from a single log stream
+// There's a many-to-one relationship between clients and pipelines.
+// Each container with the same config will get its own client to the same pipeline.
+type ClientLogger struct {
+	logFile      io.ReadWriteCloser
+	client       beat.Client
+	pipelineHash string
+}
+
+// newClientFromPipeline creates a new Client logger with a FIFO reader and beat client
+func newClientFromPipeline(pipeline *pipeline.Pipeline, file, hashstring string) (*ClientLogger, error) {
+	// setup the beat client
+	settings := beat.ClientConfig{
+		WaitClose: 0,
+	}
+	settings.ACKCount = func(n int) {
+		logp.Info("Pipeline client (%s) ACKS; %v", file, n)
+	}
+	settings.PublishMode = beat.DefaultGuarantees
```

Comment:
Have you considered an alternative non-blocking mode?

Reply:
...There's a non-blocking mode?

Reply:
See: https://github.com/elastic/beats/blob/master/libbeat/beat/pipeline.go#L176